x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedUpdate.java

```diff
@@ -384,7 +384,7 @@ public DatafeedConfig apply(DatafeedConfig datafeedConfig, Map<String, String> h
             builder.setRuntimeMappings(runtimeMappings);
         }
         if (headers.isEmpty() == false) {
-            builder.setHeaders(filterSecurityHeaders(headers));
+            builder.setHeaders(ClientHelper.getPersistableSafeSecurityHeadersForVersion(headers, minNodeVersion));
```

Comment:
Ideally, authentication headers are already filtered and converted to minNodeVersion when persisted as part of a task configuration. When they are retrieved and used, there is no need to filter and convert them again. But it is safe to do it again with very little cost. It may also help headers that are persisted before this change.